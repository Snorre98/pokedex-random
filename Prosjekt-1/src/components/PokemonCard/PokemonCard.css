.cardContainer{
    display: grid;
    gap: 0.5rem 2rem;
    /* this will be changed to constant */
    background-color:#cfb389;
    grid-template-columns: repeat(2);
    /* grid-auto-rows: (auto, auto); */
    padding: 1rem;
    border-radius: 1.5rem;
    /* align-items: center; */
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.485), 0 6px 20px 0 rgba(0, 0, 0, 0.485);
}
.cardContainer > *{
    color:black;
    padding: 0;
    margin: 0;
}
.pokemonSprite{
    object-fit: cover;
    object-position: center;
    height: 100px;
    width: 100px;
    grid-column: 1;
    grid-row: 2;
    /* border: 2px solid slategray; */
    border-radius:  0 1.5em 0 1.5em;
    /* box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.485), 0 6px 20px 0 rgba(0, 0, 0, 0.485); */
}
.typePil{
    color: rgb(1, 66, 1);
    width: fit-content;
    padding: 0 0.5rem 0 0.5em;
    border-radius:1.5rem;
    grid-column: 2;
    grid-row: 2;
    justify-self: end;
    align-self:last baseline;
    background-color: rgb(55, 143, 55);
    border: 2px solid rgb(47, 124, 47);
    font-weight: bold;
}
.nameTitle{
    font-size: 1.5rem;
    font-weight: bold;
    grid-column: 1;
    grid-row: 1;
    align-self: start;
    /* padding: 0; */
}
.heartContainer{
    justify-self: last baseline;
    align-self:start;
    grid-column: 2;
    grid-row: 1;
    width: fit-content;
}

