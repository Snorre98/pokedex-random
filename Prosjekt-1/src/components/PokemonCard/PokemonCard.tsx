import HeartComponent from '../HeartComponent/HeartComponent';
import './PokemonCard.css';

type PokemonCardProps = {
  name: string;
  type: string;
  imgURL: string;
};

export function PokemonCard({ name, type, imgURL }: PokemonCardProps) {
  const imgAltText = name + ' sprite';

  const pokemonSprite = <img className="pokemonSprite" src={imgURL} alt={imgAltText} />;

  const typePil = <span className="typePil">{type}</span>;

  const nameTitle = <h1 className="nameTitle">{name}</h1>;

  const navigateToDetails = () => {
    window.open('/details/' + name, '_blank');
  };
  const detailsButton = (
    <button className="detailsButton" onClick={navigateToDetails} data-testid={'test-detailsBtn'}>
      Details
    </button>
  );
  const heart = (
    <div className="heartContainer">
      <HeartComponent />
    </div>
  );

  return (
    <div className="cardContainer">
      {pokemonSprite}
      {typePil}
      {nameTitle}
      {detailsButton}
      {heart}
    </div>
  );
}
export default PokemonCard;
