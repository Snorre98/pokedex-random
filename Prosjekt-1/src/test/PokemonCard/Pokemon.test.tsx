import { render, screen } from '@testing-library/react';
import PokemonCard from '../../components/PokemonCard/PokemonCard';
import catadoze from '/src/assets/catadoze.jpg';

describe('PokemonCard-test', () => {
  let mockSetRefresh: (refresh: boolean) => void;
  let mockRefresh: boolean;
  beforeEach(() => {
    //mock localstorage
    const mockLocalStorage = {
      getItem: vi.fn().mockReturnValue('[]'),
      setItem: vi.fn(),
      clear: vi.fn(),
    };

    Object.defineProperty(window, 'localStorage', {
      value: mockLocalStorage,
    });
  });
  mockRefresh = false;
  // eslint-disable-next-line prefer-const
  mockSetRefresh = vi.fn((newRefresh) => {
    mockRefresh = newRefresh;
  });
  const pokemonCard = (
    <PokemonCard name="Picachu" type="Electric" imgURL={catadoze} setRefresh={mockSetRefresh} refresh={mockRefresh} />
  );
  it('Should find props', () => {
    render(pokemonCard);
    expect(screen.getByText('Picachu')).toBeInTheDocument();
    expect(screen.getByText('Electric')).toBeInTheDocument();
    expect(screen.getByRole('img')).toHaveAttribute('src', '/project1/src/assets/catadoze.jpg');
  });
  it('has focus (button)', () => {
    render(pokemonCard);
    const button = screen.getByTestId('test-detailsBtn');
    button.focus();
    expect(button).toHaveFocus();
  });
  it('PokemonCard snapshot', () => {
    const result = render(pokemonCard);
    expect(result).toMatchSnapshot();
  });
});
