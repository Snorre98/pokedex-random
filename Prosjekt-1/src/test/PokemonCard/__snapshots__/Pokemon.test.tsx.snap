// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PokemonCard-test > PokemonCard snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cardContainer"
      >
        <img
          alt="Picachu sprite"
          class="pokemonSprite"
          src="/project1/src/assets/catadoze.jpg"
        />
        <span
          class="typePil"
        >
          Electric
        </span>
        <h1
          class="nameTitle"
        >
          Picachu
        </h1>
        <button
          class="detailsButton"
          data-testid="test-detailsBtn"
        >
          Details
        </button>
        <div
          class="heartContainer"
        >
          <svg
            class="notFav"
            data-testid="heartTestID"
            viewBox="0 0 32 29.6"
          >
            <filter
              id="inset-shadow"
            >
              <feoffset
                dx="0"
                dy="0"
              />
              <fegaussianblur
                result="offset-blur"
                stdDeviation="1"
              />
              <fecomposite
                in="SourceGraphic"
                in2="offset-blur"
                operator="out"
                result="inverse"
              />
              <feflood
                flood-color="black"
                flood-opacity=".95"
                result="color"
              />
              <fecomposite
                in="color"
                in2="inverse"
                operator="in"
                result="shadow"
              />
              <fecomposite
                in="shadow"
                in2="SourceGraphic"
                operator="over"
              />
            </filter>
            <path
              d="M23.6,0c-3.4,0-6.3,2.7-7.6,5.6C14.7,2.7,
    11.8,0,8.4,0C3.8,0,0,3.8,0,8.4c0,9.4,9.5,11.9,16,
    21.2c6.1-9.3,16-12.1,16-21.2C32,3.8,28.2,0,23.6,0z"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="cardContainer"
    >
      <img
        alt="Picachu sprite"
        class="pokemonSprite"
        src="/project1/src/assets/catadoze.jpg"
      />
      <span
        class="typePil"
      >
        Electric
      </span>
      <h1
        class="nameTitle"
      >
        Picachu
      </h1>
      <button
        class="detailsButton"
        data-testid="test-detailsBtn"
      >
        Details
      </button>
      <div
        class="heartContainer"
      >
        <svg
          class="notFav"
          data-testid="heartTestID"
          viewBox="0 0 32 29.6"
        >
          <filter
            id="inset-shadow"
          >
            <feoffset
              dx="0"
              dy="0"
            />
            <fegaussianblur
              result="offset-blur"
              stdDeviation="1"
            />
            <fecomposite
              in="SourceGraphic"
              in2="offset-blur"
              operator="out"
              result="inverse"
            />
            <feflood
              flood-color="black"
              flood-opacity=".95"
              result="color"
            />
            <fecomposite
              in="color"
              in2="inverse"
              operator="in"
              result="shadow"
            />
            <fecomposite
              in="shadow"
              in2="SourceGraphic"
              operator="over"
            />
          </filter>
          <path
            d="M23.6,0c-3.4,0-6.3,2.7-7.6,5.6C14.7,2.7,
    11.8,0,8.4,0C3.8,0,0,3.8,0,8.4c0,9.4,9.5,11.9,16,
    21.2c6.1-9.3,16-12.1,16-21.2C32,3.8,28.2,0,23.6,0z"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
